app/models/agents/telegram_agent.rb

```diff
@@ -134,23 +125,52 @@ def load_file(url)
       file
     end
 
-    def unlink_file(file)
-      file.close
-      file.unlink
+    def receive_event(event)
+      interpolate_with event do
+        messages_send = TELEGRAM_ACTIONS.count do |field, _method|
+          payload = load_field event, field
+          next unless payload
+          send_telegram_messages field, configure_params(field => payload)
+          unlink_file payload if payload.is_a? Tempfile
+          true
+        end
+        error("No valid key found in event #{event.payload.inspect}") if messages_send.zero?
+      end
     end
 
-    private
+    def send_message(field, params)
+      response = HTTMultiParty.post telegram_bot_uri(TELEGRAM_ACTIONS[field]), query: params
+      unless response['ok']
+        error(response)
+      end
+    end
 
-    def configure_params(params)
-      params[:disable_notification] = interpolated['disable_notification'] if interpolated['disable_notification'].present?
-      if params.has_key?(:text)
-        params[:disable_web_page_preview] = interpolated['disable_web_page_preview'] if interpolated['disable_web_page_preview'].present?
-        params[:parse_mode] = interpolated['parse_mode'] if interpolated['parse_mode'].present?
+    def send_telegram_messages(field, params)
+      if !interpolated['long_message'].present? || interpolated['long_message'] != 'split'
+        params[:caption] = params[:caption][0..199] if params[:caption]
+        params[:text] = params[:text][0..4095] if params[:text]
+        send_message field, params
+      elsif field == :text
+        params[:text].scan(/\G(?:\w{4096}|.{1,4096}(?=\b|\z))/m) do |message|
```

Comment:
I don't understand the expression 100%, it is trying to split up the text by words first and only falling back to breaking words if it has to, right?

Can you add a spec that would fail with a simpler version like `/.{4096}/m`?

Reply:
I hope so, regular expressions have never been my cup of tea.
Anyway, I rewrote the test to verify word and non-word splitting.